src/libraries/Common/src/SourceGenerators/SourceWriter.cs

```diff
@@ -110,11 +110,11 @@ private static ReadOnlySpan<char> GetNextLine(ref ReadOnlySpan<char> remainingTe
             return next;
         }
 
-        private static unsafe void AppendSpan(StringBuilder builder, ReadOnlySpan<char> span)
+        private static void AppendSpan(StringBuilder builder, ReadOnlySpan<char> span)
         {
-            fixed (char* ptr = span)
+            foreach (char c in span)
             {
-                builder.Append(ptr, span.Length);
+                builder.Append(c);
```

Comment:
Question: do we need to use various Unsafe optimizations for SourceGens inside dotnet/runtime? I doubt they improve build time by any measurable matter

Reply:
I see. it's unfortunate there is no Append(Span) overload in NETStandard 😞

Reply:
Yeah. I think @dotnet/roslyn are considering a source writing class that also handles string interpolation. I think we could get rid of the unsafe code once we move to that.